micrometer-core/src/main/java/io/micrometer/core/instrument/binder/jetty/JettyServerThreadPoolMetrics.java

```diff
@@ -41,49 +46,80 @@
         this.threadPool = threadPool;
         this.tags = tags;
     }
 
     @Override
     public void bindTo(MeterRegistry registry) {
+        this.registry = registry;
         if (threadPool instanceof SizedThreadPool) {
             SizedThreadPool sizedThreadPool = (SizedThreadPool) threadPool;
-            Gauge.builder("jetty.threads.config.min", sizedThreadPool, SizedThreadPool::getMinThreads)
+            Gauge minGauge = Gauge.builder(MIN, sizedThreadPool, SizedThreadPool::getMinThreads)
                 .description("The minimum number of threads in the pool")
                 .tags(tags)
                 .register(registry);
-            Gauge.builder("jetty.threads.config.max", sizedThreadPool, SizedThreadPool::getMaxThreads)
+            registeredMeterIds.add(minGauge.getId());
+            Gauge maxGauge = Gauge.builder(MAX, sizedThreadPool, SizedThreadPool::getMaxThreads)
                 .description("The maximum number of threads in the pool")
                 .tags(tags)
                 .register(registry);
+            registeredMeterIds.add(maxGauge.getId());
             if (threadPool instanceof QueuedThreadPool) {
                 QueuedThreadPool queuedThreadPool = (QueuedThreadPool) threadPool;
-                Gauge.builder("jetty.threads.busy", queuedThreadPool, QueuedThreadPool::getBusyThreads)
+                Gauge busyGauge = Gauge.builder(BUSY, queuedThreadPool, QueuedThreadPool::getBusyThreads)
                     .description("The number of busy threads in the pool")
                     .tags(tags)
                     .register(registry);
-                Gauge.builder("jetty.threads.jobs", queuedThreadPool, QueuedThreadPool::getQueueSize)
+                registeredMeterIds.add(busyGauge.getId());
+                Gauge jobsGauge = Gauge.builder(JOBS, queuedThreadPool, QueuedThreadPool::getQueueSize)
                     .description("Number of jobs queued waiting for a thread")
                     .tags(tags)
                     .register(registry);
+                registeredMeterIds.add(jobsGauge.getId());
             }
         }
-        Gauge.builder("jetty.threads.current", threadPool, ThreadPool::getThreads)
+        Gauge currentGauge = Gauge.builder(CURRENT, threadPool, ThreadPool::getThreads)
             .description("The total number of threads in the pool")
             .tags(tags)
             .register(registry);
-        Gauge.builder("jetty.threads.idle", threadPool, ThreadPool::getIdleThreads)
+        registeredMeterIds.add(currentGauge.getId());
+        Gauge idleGauge = Gauge.builder(IDLE, threadPool, ThreadPool::getIdleThreads)
             .description("The number of idle threads in the pool")
             .tags(tags)
             .register(registry);
+        registeredMeterIds.add(idleGauge.getId());
+    }
+
+    @Override
+    public void close() throws Exception {
+        if (registry != null) {
+            registeredMeterIds.forEach(registry::remove);
```

Comment:
If we think that things will be reused and these meters need to be removed from the registry (reading the issue I'm still not sure or I missed something) we should definitely empty `registeredMeterIds` too.

Reply:
I don't think it would be expected for an instance of `JettyServerThreadPoolMetrics` to be reused, though I suppose it is theoretically possible. In `InstrumentedQueuedThreadPool`, it will create a new instance in `doStart`.

Reply:
I agree that an instance of JettyServerThreadPoolMetrics should not be reused, but emptying `registeredMeterIds` seems like a good idea just to avoid potential footguns. I'll make that change

Reply:
This is done